src/lib/components/linetimeseriechart/linetimeseriechart.component.tsx

```diff
@@ -0,0 +1,549 @@
+        rechartsData[timestamp][label] = value;
+      });
+    });
+
+    // Convert object to array for Recharts
+    return Object.values(rechartsData).sort(
+      (
+        a: { timestamp: number } & Record<string, string | number | null>,
+        b: { timestamp: number } & Record<string, string | number | null>,
+      ) => (a.timestamp as number) - (b.timestamp as number),
+    );
+  }, [series, startingTimeStamp, duration, frequency, yAxisType]);
+
+  // Calculate 5 perfectly evenly spaced ticks
+  const xAxisTicks = useMemo(() => {
+    if (!chartData || chartData.length === 0) return [];
+
+    const timestamps: number[] = chartData.map((d) => d.timestamp);
+    const minTimestamp = Math.min(...timestamps);
+    const maxTimestamp = Math.max(...timestamps);
+
+    // Calculate 5 perfectly evenly spaced ticks
+    const timeRange = maxTimestamp - minTimestamp;
+    const interval = timeRange / 4; // 4 intervals create 5 points
+
+    const exactEvenTicks = [
+      minTimestamp,
+      minTimestamp + interval,
+      minTimestamp + interval * 2,
+      minTimestamp + interval * 3,
+      maxTimestamp,
+    ];
+
+    // Return perfectly even ticks (guaranteed to be evenly divided)
+    return exactEvenTicks;
+  }, [chartData]);
+
+  const { topValue, unitLabel } = useMemo(() => {
+    if (yAxisType === 'percentage') return { topValue: 100, unitLabel: '%' };
+    if (!unitRange || !chartData.length) return { topValue: 0, unitLabel: '' };
+
+    const values = chartData.flatMap((dataPoint) =>
+      Object.entries(dataPoint)
+        .filter(([key]) => key !== 'timestamp')
+        .map(([_, value]) => {
+          const num =
+            typeof value === 'string' ? Number(value) : (value ?? Infinity);
+          return !isNaN(num) && num !== null ? num : null;
+        })
+        .filter((value): value is number => value !== null),
+    );
+
+    const top = Math.abs(Math.max(...values));
+    const bottom = Math.abs(Math.min(...values));
+    const maxValue = Math.max(top, bottom);
+    const roundedMax = Math.ceil(maxValue / 10) * 10;
+
+    return {
+      topValue: roundedMax,
+      unitLabel: getUnitLabel(unitRange, maxValue).unitLabel,
+    };
+  }, [chartData, yAxisType, unitRange]);
+
+  // Group series by resource and create color mapping
+  const { colorMapping, groupedSeries } = useMemo(() => {
+    const mapping: Record<string, string> = {};
+    const allSeries = isSymmetricalSeries(series)
+      ? [...series.above, ...series.below]
+      : (series as Serie[]);
+
+    // Group series by resource
+    const groups = allSeries.reduce(
+      (acc, serie) => {
+        const key = serie.resource;
+        if (!acc[key]) {
+          acc[key] = [];
+        }
+        acc[key].push(serie);
+        return acc;
+      },
+      {} as Record<string, Serie[]>,
+    );
+
+    // Assign colors to unique resources
+    Object.keys(groups).forEach((resource, index) => {
+      const color = colorRange[index % colorRange.length];
```

Comment:
There is the issue of one chart missing one serie and messing the colors

Reply:
We will address this issue through context in the following PR.